src/serde_obkv/value/mod.rs

```diff
@@ -550,6 +715,59 @@ impl Value {
         }
     }
 
+    pub fn table_obj_decode(buf: &mut BytesMut, table_obj_type: TableObjType) -> Result<Value> {
+        let obj_type = table_obj_type.to_obj_type();
+        let mut meta = ObjMeta::default_obj_meta(obj_type.clone());
+
+        match obj_type {
+            ObjType::Null => Ok(Value::default()),
+            ObjType::TinyInt => Ok(Value::Int8(decode_i8(buf)?, meta)),
+            ObjType::SmallInt => Ok(Value::Int32(decode_vi32(buf)?, meta)),
+            ObjType::Int32 => Ok(Value::Int32(decode_vi32(buf)?, meta)),
+            ObjType::Int64 => match table_obj_type {
+                TableObjType::Min => Ok(Value::Int64(-2, meta)),
+                TableObjType::Max => Ok(Value::Int64(-3, meta)),
+                _ => Ok(Value::Int64(decode_vi64(buf)?, meta)),
+            },
+            ObjType::UTinyInt => Ok(Value::UInt8(decode_u8(buf)?, meta)),
+            ObjType::USmallInt => Ok(Value::UInt32(decode_vi32(buf)? as u32, meta)),
+            ObjType::UMediumInt => Ok(Value::UInt32(decode_vi32(buf)? as u32, meta)),
+            ObjType::UInt32 => Ok(Value::UInt32(decode_vi32(buf)? as u32, meta)),
+            ObjType::UInt64 => Ok(Value::UInt64(decode_vi64(buf)? as u64, meta)),
+            ObjType::Float => Ok(Value::Float(decode_f32(buf)?, meta)),
+            ObjType::Double => Ok(Value::Double(decode_f64(buf)?, meta)),
+            ObjType::UFloat => Ok(Value::Float(decode_f32(buf)?, meta)),
+            ObjType::UDouble => Ok(Value::Double(decode_f64(buf)?, meta)),
+            //FIXME date and time
+            ObjType::DateTime => {
+                meta = ObjMeta::decode(buf)?;
+                Ok(Value::Time(decode_vi64(buf)?, meta))
+            }
+            ObjType::Timestamp => {
+                meta = ObjMeta::decode(buf)?;
+                Ok(Value::Time(decode_vi64(buf)?, meta))
+            }
+            ObjType::Date => unimplemented!(),
+            ObjType::Time => unimplemented!(),
+            ObjType::Year => unimplemented!(),
+            ObjType::Varchar => {
+                if table_obj_type == TableObjType::Varbinary {
+                    meta.cs_type = CollationType::Binary
+                }
+                Self::decode_binary(buf, meta)
+            }
+            ObjType::Char => Self::decode_binary(buf, meta),
+            // TODO: ObjType::HexString
+            ObjType::Extend => Ok(Value::Int64(decode_vi64(buf)?, meta)),
+            ObjType::TinyText => Self::decode_binary(buf, meta),
+            ObjType::Text => Self::decode_binary(buf, meta),
+            ObjType::MediumText => Self::decode_binary(buf, meta),
+            ObjType::LongText => Self::decode_binary(buf, meta),
+            ObjType::Bit => Ok(Value::Int64(decode_vi64(buf)?, meta)),
+            _ => Err(Error::Custom("Unsupported obj type.".into())),
```

Comment:
Avoid the default match arm.

Reply:
Since we return an error, I guess we could use default pattern here